### ISSUE TITLE
[Bug-Candidate]: Echidna can get stuck while shrinking

### ISSUE THREAD

**Issue by rappie** [crytic/echidna#1224]
### Describe the issue:

In certain cases, Echidna can get stuck while shrinking.

Example below:
![image](https://github.com/crytic/echidna/assets/1430820/69e4b5c7-725d-43cf-b653-a2c9d06f7ef5)

When setting `maxBlockDelay` and `maxTimeDelay` to `1`, it is easily shrunk further:
![image](https://github.com/crytic/echidna/assets/1430820/48331f5f-ee25-440f-b37e-07271dde735b)

After further research I was able to determine that timestamps/blocks actually don't play a role in the breaking of this invariant at all. So it should have been able to shrink it further.

### Code example to reproduce the issue:

The code is private, I can share the repo if you DM me on Slack

### Version:

Echidna 2.2.3 (custom build from `master`)

**Comment by ggrieco-tob** [comment]
I think we need to be more aggressive reducing the block numbers? In this case, this required at least a block each operation?

**Comment by rappie** [comment]
No, there are no block/timestamp increments necessary at all to break this invariant.

Besides block/timestamp shrinking being slow or getting stuck, there are also uneccesary transactions that don't get shrunk away. It seems to me that there is some deeper issue going on.

**Comment by ggrieco-tob** [comment]
@rappie can you please re-test the shrinking using the latest `master`?

**Comment by rappie** [comment]
Still the same issue :disappointed:

**Comment by elopez** [comment]
@rappie can you confirm if this is fixed on 2.2.7 or current master?

**Comment by rappie** [comment]
Will check this soon 🫡

**Comment by gustavo-grieco** [comment]
Any updates? 🤔

**Comment by rappie** [comment]
Not yet sadly. I've been super busy but it is still on my todo list!

**Comment by rappie** [comment]
@elopez @gustavo-grieco Tested this and it is indeed fixed 🎉

**Comment by gustavo-grieco** [comment]
Awesome, let's close this one. Thanks to everyone involved to close this ticket!
